src/Tools/GenerateGlobalAnalyzerConfigs/Program.cs

```diff
@@ -477,6 +477,22 @@ static void AddMSBuildContentForItemOptions(StringBuilder builder)
                 }
             }
 
+            static string GetCodeAnalysisTreatWarningsAsErrorsTargetContents()
+            {
+                return $@"
+  <!--
+    Design-time target to prevent the rule ids implemented in this package to be bumped to errors in the IDE
```

Comment:
It seems design time builds in IDE need this setting to be inside a target to get applied.